### ISSUE TITLE
False positive E0741

### ISSUE THREAD

**Issue by PizzaCrust** [intellij-rust/intellij-rust#9420]
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4904-222-nightly
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2022.2.2 Community Edition (IC-222.4167.29)
* **Operating system:** Windows 10 10.0
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Says that  `usize` doesn't derive both `PartialEq` and `Eq`

## Steps to reproduce
```rust
trait A {

}

impl<T, const N: usize> A for [T; N] {

}
```
